ui/lib/kv/addon/components/kv-list-filter.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import keys from 'core/utils/key-codes';
+import { keyIsFolder, parentKeyForKey, keyWithoutParentKey } from 'core/utils/key-utils';
+import escapeStringRegexp from 'escape-string-regexp';
+import { tracked } from '@glimmer/tracking';
+
+/**
+ * KvListFilter component is similar to the NavigateInput component but meets the specific routing needs for the KV Ember engine.
```

Comment:
Maybe let's describe this outside of NavigateInput (since we eventually want to remove that)